public/app/features/dashboard/components/ShareModal/ShareLink.tsx

```diff
@@ -58,19 +58,18 @@ export class ShareLink extends PureComponent<Props, State> {
     }
   }
 
-  buildUrl = () => {
+  buildUrl = async () => {
     const { panel } = this.props;
     const { useCurrentTimeRange, includeTemplateVars, useShortUrl, selectedTheme } = this.state;
 
     const shareUrl = buildShareUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
     const imageUrl = buildImageUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
 
     if (useShortUrl) {
-      getBackendSrv()
-        .post(`/api/short-urls`, {
-          path: getRelativeURLPath(shareUrl),
-        })
-        .then(res => this.setState({ shareUrl: res.url, imageUrl }));
+      const shortLink = await createShortLink(shareUrl);
+      if (shortLink) {
+        this.setState({ shareUrl: shortLink, imageUrl });
+      }
```

Comment:
Hmm, this logic might be better in the `buildShareUrl` function, WDYT?

Reply:
Great point, I totally agree. I have moved `createShortLink` to `buildShareUrl`.